glossary.md

```diff
@@ -0,0 +1,172 @@
+_Module 2_
+
+**Functional/non-functional requirements** - Functional requirements pertain to features of an application.
+Non-functional requirements (NFRs) describe other attributes of a system, such as performance, accessibility and security.
+NFRs are often not explicitly written down on tickets, but are nonetheless critical to building a working application.  
+_Module 6_
+
+**IaC** - Infrastructure as Code, managing the provisioning of Cloud infrastructure as code rather than through manual processes.
+Examples of Infrastructure as Code tools are Terraform, Pulumi, ARM templates and CloudFormation.  
+_Module 12_
+
+**Immutable Infrastructure** - Treating servers as immutable, that is they are only created and destroyed, never updated in place.
+This ensures your servers are recycled with each release, which makes it easier to know what state they are in and can allow for easier rollbacks if you leave the old servers up until each release is complete.  
+_Module 5_
+
+**Impact mapping** - A diagram connecting business goals to deliverable tasks, see <https://www.impactmapping.org/drawing.html>.  
+_Module 6_
+
+**Metrics** - Properties of a service that you monitor over time. Could be CPU or Memory usage, the number of requests or errors, or a custom metric that's specific to your application.  
+_Module 13_
+
+**Microservices** - An architectural pattern using lots of small services to form a system rather than one large monolith.
+
+**Mob programming** - Programming with a group (>2, that would be Pair Programming) of people.  
+_Module 2_
+
+**MTBF** - "Mean Time Between Failures", a measure of how often a service fails. MTTF "Mean Time To Failure" is a similar measure for equipment that eventually fails permanently, like a hard drive, though they are sometimes used interchangeably when talking about services.  
+_Module 6_
+
+**MTTR** - "Mean Time To Recovery", how long it takes to fix critical system failures.
+DevOps principals encourage aiming to reduce MTTR over increasing MTBF as a better route to moving quickly with low downtime.  
+_Module 6_
+
+**Penetration testing** - AKA Pen testing. Evaluating the security of a system by attempting to breach it.  
+_Module 10_
+
+**Patching** - Applying "patches" (updates) to a server/service, which might be to the operating system, components like web servers or direct library dependencies.
+It's important to apply security patches quickly when they become available.
+CD, Immutable infrastructure and containerisation can all help make easier.
+Tools like Dependabot or Renovate can automate raising PRs to update dependencies.  
+_Module 5_
+
+**Pipeline** - An automated CI/CD process. The pipeline analogy comes from having several steps run in sequence, like passing through a pipe.  
+_Module 7_
+
+**Post-Mortem** - A review session after an incident to determine what went wrong and how to prevent a repeat.
+Sometimes called a retro or retrospective.  
```

Comment:
Is this true? I think of them as fairly distinct things, but I know Phil disagreed with me too when I said that
In my head, retro is for a process/period of time, whereas post-mortem is specifically for an unplanned incident